WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailViewModel.kt

```diff
@@ -24,10 +59,78 @@ class CategoryDetailViewModel @Inject constructor(
         dispatcher.unregister(this)
     }
 
-    fun start(siteModel: SiteModel) {
+    fun start() {
         if (isStarted) return
         isStarted = true
 
-        this.siteModel = siteModel
+        initCategories()
+    }
+
+    private fun initCategories() {
+        launch {
+            val siteCategories = getCategoriesUseCase.getSiteCategories(siteModel)
+            siteCategories.add(0, topLevelCategory)
+            _uiState.postValue(
+                    UiState(
+                            categories = siteCategories,
+                            selectedParentCategoryPosition = 0,
+                            categoryName = ""
+                    )
+            )
+        }
+    }
+
+    fun onSubmitButtonClick() {
+        _uiState.value?.let { state ->
+            addCategory(state.categoryName, state.categories[state.selectedParentCategoryPosition])
+        }
+    }
+
+    private fun addCategory(categoryText: String, parentCategory: CategoryNode) {
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            _onCategoryPush.postValue(
+                    Event(Failure(UiStringRes(R.string.no_network_message)))
+            )
+            return
+        }
+        launch {
+            _onCategoryPush.postValue(Event(InProgress))
+            addCategoryUseCase.addCategory(categoryText, parentCategory.categoryId, siteModel)
+        }
+    }
+
+    fun onCategoryNameUpdated(inputValue: String) {
+        uiState.value?.let { state ->
+            val submitButtonUiState = if (inputValue.isNotEmpty()) {
+                SubmitButtonEnabledUiState
+            } else {
+                SubmitButtonDisabledUiState
+            }
+            _uiState.value = state.copy(
+                    categoryName = inputValue,
+                    submitButtonUiState = submitButtonUiState
+            )
+        }
+    }
+
+    fun onParentCategorySelected(position: Int) {
+        _uiState.value?.let { state ->
+            _uiState.value = state.copy(selectedParentCategoryPosition = position)
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = MAIN)
+    fun onTermUploaded(event: OnTermUploaded) {
+        if (event.isError) AppLog.e(
+                T.SETTINGS,
+                "An error occurred while uploading taxonomy with type: " + event.error.type
+        )
+        val categoryUiState = if (event.isError) {
+            Failure(UiStringRes(string.adding_cat_failed))
```

Comment:
🔍 

1. `CategoryDetailViewModel`: Missing `R` qualifier
`string.adding_cat_failed`
`string.adding_cat_success`

2. `CategoryDetailFragment`: Missing `R` qualifier
`layout.categories_row_parent`

3. `CategoryDetailViewModelTest`: `R.string` import can be removed.

Reply:
@ashiagr Nice catch. Done in a2218ca. 👍